### PR TITLE
Feat/completed generated toc script

### PR DESCRIPTION
**Pull request by thboileau**

### CODE REVIEW

File: build/tools/generate-toc.py [hunk 1]
```diff
@@ -99,18 +108,64 @@ def add_markdown_file(path, document, yml_path, path_to_display, link_url):
...
         print '%s* %s' % (indentation, line['title'])
         for item in line['items']:
-            scan_line(item, indentation + '  ')
+            print_line_as_markdown(item, indentation + '  ')
     elif 'file' in line.keys():
         # Is file
         print '%s* [%s](%s)' % (indentation, line['title'], line['link_url'])
 
+def convert_to_dhc_json_index(document):
+    indentation = '    '
```
Comment (guiblondeau) [inline build/tools/generate-toc.py]: indentation doesn't seem useful

**Comment by guiblondeau** [comment]
I'd be interested to see what "dhcjsonindex" looks like

File: build/tools/generate-toc.py [hunk 2]
```diff
@@ -26,10 +30,14 @@ def main():
 
     obj = {'toc': document}
 
-    if not markdown:
```
Comment (guiblondeau) [inline build/tools/generate-toc.py]: Oh I did it a long time ago ... I would never have done this today ^^

File: build/tools/generate-toc.py [hunk 3]
```diff
@@ -99,18 +108,64 @@ def add_markdown_file(path, document, yml_path, path_to_display, link_url):
...
-            scan_line(item, indentation + '  ')
+            print_line_as_markdown(item, indentation + '  ')
     elif 'file' in line.keys():
         # Is file
         print '%s* [%s](%s)' % (indentation, line['title'], line['link_url'])
 
+def convert_to_dhc_json_index(document):
+    indentation = '    '
+    listHelpPages = []
+    listHelpPages.append('{"dir": false, "title":"%s", "path": "%s"}' % ("User Guide", "/index"))
```
Comment (guiblondeau) [inline build/tools/generate-toc.py]: Just wondering is a dictionary converted to JSON would be more readable. eg:
```python
help_page = {
  'dir': false,
  'title': 'User guide',
  'path': '/index'
}
listHelpPages.append(json.dumps(help_page));
```

Never mind, keep it like that, it's just the toc script

**Comment by guiblondeau** [comment]
LGTM

File: build/tools/generate-toc.py (hunk 1 above)
Comment (thboileau) [inline build/tools/generate-toc.py]: sure, removed.

**Comment by thboileau** [comment]
Hi @guiblondeau ,

here is a sample json index:

```json
[
{"dir": false, "title":"User Guide", "path": "/index"},
{"dir": true, "title":"Get started", "path": "/get-started"},
{"dir": false, "title":"Launch DHC", "path": "/get-started/install"},
{"dir": false, "title":"Sign-in", "path": "/get-started/sign-in"},
{"dir": false, "title":"Try it for FREE", "path": "/get-started/free-trial"},
{"dir": false, "title":"Subscribe to a plan", "path": "/get-started/subscribe"},
{"dir": false, "title":"Access account info", "path": "/get-started/account-info"},
{"dir": true, "title":"Perform requests", "path": "/perform-requests"},
{"dir": false, "title":"Build requests", "path": "/perform-requests/build-request"},
{"dir": false, "title":"Display responses", "path": "/perform-requests/display-response"},
{"dir": true, "title":"Use assertions", "path": "/perform-requests/assertions"},
{"dir": false, "title":"Overview", "path": "/perform-requests/assertions/overview"},
{"dir": false, "title":"Assertions basics", "path": "/perform-requests/assertions/assertions-basics"},
{"dir": false, "title":"Response parts", "path": "/perform-requests/assertions/response-parts"},
{"dir": false, "title":"JSON & XML payloads", "path": "/perform-requests/assertions/json-payloads"},
{"dir": false, "title":"Expressions in assertions", "path": "/perform-requests/assertions/expressions"},
{"dir": false, "title":"Work with History", "path": "/perform-requests/history"},
{"dir": true, "title":"Use expressions", "path": "/perform-requests/expressions"},
{"dir": false, "title":"Overview", "path": "/perform-requests/expressions/overview"},
{"dir": false, "title":"References", "path": "/perform-requests/expressions/references"},
{"dir": false, "title":"Built-in methods", "path": "/perform-requests/expressions/built-in-methods"},
{"dir": false, "title":"Grammar", "path": "/perform-requests/expressions/grammar"},
{"dir": false, "title":"Use context attributes", "path": "/perform-requests/context-attributes"},
{"dir": true, "title":"Manage your drive", "path": "/drive"},
{"dir": false, "title":"Create new item", "path": "/drive/create-new-item"},
{"dir": false, "title":"Organize requests", "path": "/drive/organize-requests"},
{"dir": true, "title":"Import & export requests", "path": "/drive/import-export"},
{"dir": false, "title":"Import projects", "path": "/drive/import-export/import-projects"},
{"dir": false, "title":"Export projects", "path": "/drive/import-export/export-projects"},
{"dir": false, "title":"Data structure", "path": "/drive/import-export/data-structure"},
{"dir": false, "title":"Synchronize data", "path": "/drive/synchronize-data"},
{"dir": true, "title":"Run tests", "path": "/run-tests"},
{"dir": false, "title":"Create scenarios", "path": "/run-tests/scenarios"},
{"dir": true, "title":"Automate tests", "path": "/run-tests/automate-tests"},
{"dir": false, "title":"Get test reports", "path": "/run-tests/automate-tests/test-reports-maven"},
{"dir": false, "title":"Run tests in CI", "path": "/run-tests/automate-tests/jenkins-ci"},
{"dir": true, "title":"Share your projects", "path": "/share"},
{"dir": false, "title":"Add team members", "path": "/share/add-members"},
{"dir": false, "title":"Share your projects", "path": "/share/share-projects"}
]
```

File: build/tools/generate-toc.py (hunk 3 above)
Comment (thboileau) [inline build/tools/generate-toc.py]: yes, to my mind, this is only a set of lines of text. let's keep this simple, for now.

**Comment by guiblondeau** [comment]
Thanks for the sample :)

**Pull request merged**